src/item.cpp

```diff
@@ -8468,7 +8226,7 @@ bool item_ptr_compare_by_charges( const item *left, const item *right )
     } else if( right->contents.empty() ) {
         return true;
     } else {
-        return right->contents.front().charges < left->contents.front().charges;
+        return right->contents.legacy_front().charges < left->contents.legacy_front().charges;
```

Comment:
this needs some attention: I have no idea what this function's even for. obviously it's old code, and it's assuming only one item can be inside another item which isn't gonna work